administrator/components/com_media/tmpl/media/default.php

```diff
@@ -23,7 +23,15 @@
 HTMLHelper::_('behavior.keepalive');
 
 // Add javascripts
-HTMLHelper::_('script', 'media/com_media/js/mediamanager.js');
+//JHtml::_('script', 'media/com_media/js/mediamanager.js');
+
+Factory::getDocument()->addScriptDeclaration('
+document.addEventListener(\'WebComponentsReady\', function() {
+	var elem = document.createElement(\'script\');
+	elem.src = "' . \Joomla\CMS\Uri\Uri::root(false) . 'media/com_media/js/mediamanager.js";
```

Comment:
@kasvith add
```js
elem.setAttribute("type", "text/javascript");
```
to pass the csp

Reply:
wait

Reply:
sure

Reply:
This feels wrong. Why do we need to use WebComponentsReady?

Reply:
Why? The polyfills are loaded after DOMContentLaoded (lazyloaded with checks on the client support) thus we cannot use the old-fashioned DOM ready event. To use the DOM ready it means that we need to check the client capabilities in the server (sniffing the UA in the request) which has some very obvious limitations and for sure will be more expensive performance wise

Reply:
Yes, I know. But this is because of the way we are loading/using the polifills. 
Other solutions: 

1) (you are going to hate me for this one) deliver all polyfills that are needed for joomla min browser support by default (and let the User decide how much he wants to improve performance/drop polyfills).

2) Deliver polifills for media manager in media manager and do not expect joomla to handle this for us.

3) Rename the event name (which is imo kind of missleading) to something better. Like PolyfillsLoaded/Ready or even JoomlaLoaded/Ready, so that we decouple polyfills from webcomponents.

Reply:
Well, No 1 is the worse solution but the easiest to adopt (let's not do that)

N2 we're gonna have duplicate code one way or another because we also need these polyfills for the web components

N3 Probably this should be named as the standard Joomla event `joomla:updated` so I can see a `joomla:ready` event. Probably I need to spend some time and make a method similar to Joomla.webComponents() that would handle the loading of the polyfills (eg extend the code we already have and refactor it so we could pass the polyfill and the required check in the json, or something similar)

My point of view here is that I don't want to make polyfills a hard dependency as at some point in the life cycle of Joomla 4 IE11 will be deprecated/unused and then all these will be just bloated code...

Reply:
IE11 is supported by MS until the end of support for windows 10 which is 2025 :(

Reply:
#FML

Reply:
We can only go by the official dates ;)

Reply:
@brianteeman was that the case for IE8, IE9 and IE10? I don't remember if people stop supporting those before or after EOL

Reply:
yes we can only go by the official dates under current policy